### DISCUSSION TITLE
How to avoid requesting images of all resolutions

### DISCUSSION THREAD

**Discussion by tianzhich** [gildas-lormeau/SingleFile#1460] [Q&A]
I found an option `removeAlternativeImages` that could be used to remove images for alternative screen resolutions. However, even after enabling this option, images for all resolutions are still requested. I noticed in the code that it handles the `srcset` attribute of the images, but there is no option to disable this behavior.

**Comment by tianzhich** [comment]
This will help to improve the speed of saving with singlefile, avoid downloading images in all resolutions on websites where resource downloads are slow.

**Reply by tianzhich** [reply]
Hi @gildas-lormeau what do you think?

**Comment by gildas-lormeau** [comment]
Indeed, I agree with you that it's something to be desired. Unfortunately, from a technical point of view, when SingleFile processes `srcset` attributes, it doesn't know for sure whether it will be able to load the displayed image. That's why this issue exists. To get around it, I've added a hidden option [`blockAlternativeImages`](https://github.com/gildas-lormeau/single-file-core/commit/bd82ec57475ee1b9ff1db7cf363116092f2d75f1), which you can activate by following this procedure: https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options. This new option will be available in the next version.

**Comment by tianzhich** [comment]
> when SingleFile processes srcset attributes, it doesn't know for sure whether it will be able to load the displayed image.

Why doesn't it know? Are you referring that some image tags only have the `srcset` attribute set but not the `src` attribute as a fallback? In that case, not requesting the image resources defined on `srcset` would cause issues, right? 

If that's the case, can we check like this: similar to the [`removeAlternativeImages`](https://github.com/gildas-lormeau/single-file-core/blob/bd82ec57475ee1b9ff1db7cf363116092f2d75f1/modules/html-images-alt-minifier.js#L88C1-L91C54) option, when `blockAlternativeImages` is enabled, only block the request to the resources defined in `srcset` if the `src` attribute exists. If `src` does not exist, the resources in `srcset` will still be requested.

**Reply by gildas-lormeau** [reply]
I'm referring to the case where for some reasons the server would refuse to serve the image defined in the `src` attribute. The case you mentioned should work even with `blockAlternativeImages` set to `true`.
